### PR TITLE
Add missing `prefix_suffix_case_sensitive` to LSF

### LINKED ISSUE #10471: Cannot override LexicalSyntacticFeaturizer in 2.8.x
### Rasa Open Source version

2.8.15

### Rasa SDK version

_No response_

### Rasa X version

_No response_

### Python version

3.8

### What operating system are you using?

Linux

### What happened?

The LexicalSyntacticFeaturizer cannot be overridden in rasa 2.8.x because its [load method](https://github.com/RasaHQ/rasa/blob/3b9df765f47a863074d7a770ef72f0ebda0fcd3c/rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py#L293) always calls its parent class instead of the current class.  This has been fixed in [main](https://github.com/RasaHQ/rasa/blob/33c03422776398f1ba46c487abb2be0888c4a953/rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py#L511-L516).

Can we port the fix to 2.8.x?

### Command / Request

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by raoulvm**
**Proposed changes**:
- Allow parameter `prefix_suffix_case_sensitive` to be set to False (Defaults to True)
  - https://github.com/rasahq/rasa/issues/5759 Asks the featurizers to support for case-insensitivity, which never was implemented in LexicalSyntacticFeaturizer
- ~Fix #10471~ has been moved to https://github.com/RasaHQ/rasa/pull/10548 by @euklein
- 
**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/nlu/featurizers/test_lexical_syntactic_featurizer.py
Comment (hsm207) [inline tests/nlu/featurizers/test_lexical_syntactic_featurizer.py:121]: This line can be removed

File: tests/nlu/featurizers/test_lexical_syntactic_featurizer.py
Comment (hsm207) [inline tests/nlu/featurizers/test_lexical_syntactic_featurizer.py:94]: To test this functionality, the test should assert whether the features of "Goodbye" are identical to the features of "goodby"

**Review by hsm207 — CHANGES_REQUESTED** [review]

**Comment by hsm207** [comment]
@JEM-Mosig This feature needs to be rewritten again for 3.x, doesn't it?

File: tests/nlu/featurizers/test_lexical_syntactic_featurizer.py
Comment (raoulvm) [reply tests/nlu/featurizers/test_lexical_syntactic_featurizer.py]: Done.

File: tests/nlu/featurizers/test_lexical_syntactic_featurizer.py
Comment (raoulvm) [reply tests/nlu/featurizers/test_lexical_syntactic_featurizer.py:94]: Reworked the test to check only for different features of the same token.lower()
in https://github.com/RasaHQ/rasa/pull/10482/commits/e2e36d5d05ecfdd6caddfe33c176c9018ae6e624

**Comment by raoulvm** [comment]
> This feature needs to be rewritten again for 3.x, doesn't it?

I think so. But in Rasa 3.0 somebody had the splendid idea to define a **class** method `_extract_raw_features_from_token` which is not instance based so it cannot KNOW which function (default or lowercase) to use.  AFAIK that is only called from an instance method, so it could also be an instance method, but what do I know.

**Review by hsm207 — APPROVED** [review]

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (JEM-Mosig) [inline rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:91]:
```suggestion
        """Creates a LexicalSyntacticFeaturizer.
```

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (JEM-Mosig) [inline rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:97]:
```suggestion
            component_config: Component config as defined in config.yml
            feature_to_idx_dict: Prepared feature to idx for loading from disk
```

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (JEM-Mosig) [inline rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:322]:
```suggestion
        Returns the metadata necessary to load the model again.
```

File: changelog/10482.bugfix.md
Comment (JEM-Mosig) [inline changelog/10482.bugfix.md:2]:
```suggestion
Adds a new optional config argument `prefix_suffix_case_sensitive` which can be set to `False` to ignore cases in prefix and suffix functions.
```
The `Allows LexicalSyntacticFeaturizer to be overridden.` part should go into a separate change log file `10471.bugfix.md`. Also, this file (`10482.bugfix.md`) should be `10482.improvement.md`.

File: tests/nlu/featurizers/test_lexical_syntactic_featurizer.py
Comment (JEM-Mosig) [inline tests/nlu/featurizers/test_lexical_syntactic_featurizer.py:88]:
```suggestion
def test_text_featurizer_case_insensitive(sentence: Text, expected_features: List[float]):
```

**Review by JEM-Mosig — COMMENTED** [review]
I left a few comments.

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (raoulvm) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:91]: Won't pass flake8. Your linter is set to "very very picky"

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (raoulvm) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:320]: Won't pass flake8

File: changelog/10482.bugfix.md
Comment (raoulvm) [reply changelog/10482.bugfix.md]: Separated

**Comment by raoulvm** [comment]
@ancalita Could you check why the doc test fails? I cannot see the reason in my changes.

**Comment by ancalita** [comment]
> @ancalita Could you check why the doc test fails? I cannot see the reason in my changes.

Some flaky behaviour caused by some external links from Amazon, I've been having myself some issues with this step today.
I will just re-run this step, if it fails again, please feel free to re-run it.

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (raoulvm) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:91]: @JEM-Mosig mind to resolve?

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (raoulvm) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:320]: @JEM-Mosig mind to resolve?

File: changelog/10482.bugfix.md
Comment (raoulvm) [reply changelog/10482.bugfix.md]: @JEM-Mosig done.

**Comment by raoulvm** [comment]
Thank you @ancalita this time it ran through. Now someone "with write access" needs to click the merge button :-)

**Comment by koernerfelicia** [comment]
Hi @raoulvm, just a heads up, Johannes is now on education days, so I will be taking over the review. Sorry for the inconvenience!

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (koernerfelicia) [inline rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:54]:
```suggestion
        # sensitivity
```

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (koernerfelicia) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:91]: I can't seem to reproduce this... `make lint` runs fine for these changes. Can you let me know what commands you ran?

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (koernerfelicia) [inline rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:315]:
```suggestion
        return cls(meta, feature_to_idx_dict=feature_to_idx_dict)
```
Think we don't need the comment within the code, the changelog should be enough evidence :)

File: rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py
Comment (raoulvm) [reply rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py:91]: I used the flake8 commands locally that are part of `make lint` and it complained about the mood of the first line, should be imparative.
Commit https://github.com/RasaHQ/rasa/pull/10482/commits/9923e9f0fb3f4d8be6ccd6f451dd3ccc4ea7564d just had a green light on **passing all tests**. I will re-check by committing this into the branch. Let's see if my local flake8 was even pickier...

**Comment by raoulvm** [comment]
@koernerfelicia : Ok, it ran through. So `make lint` on my PC was stricter than the pipeline, good to know.

**Comment by raoulvm** [comment]
@koernerfelicia: I have changed that and it worked (https://github.com/RasaHQ/rasa/pull/10482/commits/17222db67446258b16a3907eb7e73593e69b23a8), but I get 2 failing checks since I merged `2.8.x` into my branch!? (https://github.com/RasaHQ/rasa/pull/10482/commits/46b3cf6aa3649a24eab49f1f9f1d36162d23e194)

**Comment by raoulvm** [comment]
@hsm207 Hi Shukri did you at Rasa drop support for python 3.6 under windows but did not remove the checks in the pipeline?

**Comment by hsm207** [comment]
> 

@raoulvm no, we did not. Can you rerun the job?

**Comment by raoulvm** [comment]
> Can you rerun the job?

How should I do that?

**Comment by euklein** [comment]
@hsm207 You mean hitting re-run all jobs on the actions tab?

**Comment by hsm207** [comment]
> @hsm207 You mean hitting re-run all jobs on the actions tab?

@euklein yes

**Comment by raoulvm** [comment]
Thanks. I seem not to have this famous button there.

**Comment by ancalita** [comment]
@hsm207 Could you please double-check this PR and confirm if your approval still stands before I merge it?

**Review by hsm207 — COMMENTED** [review]
@ancalita the code is fine but the changelog is wrong.

Issue https://github.com/RasaHQ/rasa/issues/10471 was fixed it 2.8.17.

This PR is just for to add a parameter to control whether LSF will lowercase the tokens or not.

Should this be considered a feature, improvement or a bug fix?

**Comment by ancalita** [comment]
> Should this be considered a feature, improvement or a bug fix?

Sounds like an improvement, so one changelog entry is correct, but the bugfix one can be removed.

**Comment by hsm207** [comment]
@ancalita done!

**Comment by hsm207** [comment]
> Sounds like an improvement, so one changelog entry is correct, but the bugfix one can be removed.

done!

**Comment by raoulvm** [comment]
Thank you @hsm207 - will you take care of https://github.com/RasaHQ/rasa/pull/10483 as well, so it does not get lost in 3.x?

**Pull request merged**
